ui/src/components/connections/ConnectionDetails.tsx

```diff
@@ -1,40 +1,144 @@
-import { Avatar, Button, Card, Row, Space, Tag, Typography } from "antd";
+import {
+  Avatar,
+  Button,
+  Card,
+  Radio,
+  RadioChangeEvent,
+  Row,
+  Space,
+  Tag,
+  Tooltip,
+  Typography,
+  message,
+} from "antd";
+import Table, { ColumnsType } from "antd/es/table";
+import dayjs, { extend as extendDayJsWith } from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
 import { useCallback, useEffect, useState } from "react";
 import { useParams } from "react-router-dom";
 
-import { Detail } from "../schemas/Detail";
 import { APIError } from "src/adapters/api";
 import { Connection, getConnection } from "src/adapters/api/connections";
-import { ReactComponent as IconCreditCardPlus } from "src/assets/icons/credit-card-plus.svg";
+import {
+  Credential,
+  CredentialType,
+  credentialTypeParser,
+  getCredentials,
+} from "src/adapters/api/credentials";
 import { ReactComponent as IconCreditCardRefresh } from "src/assets/icons/credit-card-refresh.svg";
 import { ReactComponent as IconTrash } from "src/assets/icons/trash-01.svg";
+import { ConnectionDetailsRowDropdown } from "src/components/connections/ConnectionDetailsRowDropdown";
+import { IssueDirectlyButton } from "src/components/connections/IssueDirectlyButton";
+import { Detail } from "src/components/shared/Detail";
 import { ErrorResult } from "src/components/shared/ErrorResult";
 import { LoadingResult } from "src/components/shared/LoadingResult";
+import { NoResults } from "src/components/shared/NoResults";
 import { SiderLayoutContent } from "src/components/shared/SiderLayoutContent";
 import { TableCard } from "src/components/shared/TableCard";
 import { useEnvContext } from "src/contexts/env";
 import { isAbortedError, makeRequestAbortable } from "src/utils/browser";
-import { IDENTIFIER, ISSUE_CREDENTIAL } from "src/utils/constants";
+import {
+  CREDENTIALS,
+  DOTS_DROPDOWN_WIDTH,
+  IDENTIFIER,
+  ISSUE_CREDENTIAL,
+  ISSUE_DATE,
+} from "src/utils/constants";
+import { processZodError } from "src/utils/error";
 import { formatDate } from "src/utils/forms";
-import { AsyncTask, isAsyncTaskDataAvailable, isAsyncTaskStarting } from "src/utils/types";
+import {
+  AsyncTask,
+  hasAsyncTaskFailed,
+  isAsyncTaskDataAvailable,
+  isAsyncTaskStarting,
+} from "src/utils/types";
+
+const obfuscateDID = (did: string) => {
+  const didSplit = did.split(":").slice(0, -1).join(":");
+  const address = did.split(":").pop();
+
+  return address
+    ? `${didSplit}${address.substring(0, 5)}...${address.substring(address.length - 5)}`
```

Comment:
Please let's consider [this option](https://0xpolygon.slack.com/archives/C03HUC0Q7RU/p1680620107749099).

Reply:
We agreed to implement this on the next PR

Reply:
Why not do it here?